print-arista.c

```diff
@@ -93,17 +93,13 @@ arista_print_date_hms_time(netdissect_options *ndo, uint32_t seconds,
 		uint32_t nanoseconds)
 {
 	time_t ts;
-	struct tm *tm;
 	char buf[BUFSIZE];
 
 	ts = seconds + (nanoseconds / 1000000000);
 	nanoseconds %= 1000000000;
-	if (NULL == (tm = gmtime(&ts)))
-		ND_PRINT("gmtime() error");
-	else if (0 == strftime(buf, sizeof(buf), "%Y-%m-%d %H:%M:%S", tm))
-		ND_PRINT("strftime() error");
-	else
-		ND_PRINT("%s.%09u", buf, nanoseconds);
+	ND_PRINT("%s",
+	    nd_format_time_with_nanoseconds(buf, sizeof(buf), "%Y-%m-%d %H:%M:%S",
+	       gmtime(&ts), nanoseconds));
```

Comment:
The same can be done simpler:
```c
	ND_PRINT("%s.%09u",
	    nd_format_time(buf, sizeof(buf), "%Y-%m-%d %H:%M:%S", gmtime(&ts)),
	    nanoseconds);
```
This way `nd_format_time_with_nanoseconds()` would not have to detect ns values greater than 10**9, would not have to squeeze the fractional part into the buffer and basically would not have reasons to exist.

Reply:
...and you'd run the risk of getting output such as

    [Does not fit in buffer].013657019

rather than just

    [Does not fit in buffer]

Reply:
...which would say "the fractional part is definitely this many nanoseconds, but the integer part is not a even a valid date, so you would need to inspect the packet closer to understand what this timestamp means".

Reply:
> "integer part is not a even a valid date"

Given the current code, that would only be the case if the integer part were a negative number with an absolute value so great that the year was before 0 CE, and I'm not sure that `[Does not fit in buffer].013657019` would clearly indicate that. `-1-01-31 00:23:59..013657019` would probably indicate that better; I've adjusted the sample string to allow negative years, as `tm_year` is the only field that doesn't have a strict limit on values.  That means that, at least for the one place that uses `nd_format_time_with_nanoseconds()`, "Does not fit in buffer" means "there's a bug in the code" (or in `localtime()` or `gmtime()`), not "the integer part isn't a valid date".

The one place where the `struct tm` isn't generated from a `time_t` is the DOS-style date processing in `interpret_dos_date()` in smbutil.c, but that keeps all fields other than `tm_year` in the range 0-99, so, again, bogus dates and times will show up as 2023-01-32 23:66:77 or something such as that, not as "Does not fit in buffer".

For `[Unrepresentable].013657019`, that is, in practice, an indication that 1) this is on Windows and Microsoft have decided not to support converting pre-UNIX Epoch times in `localtime()` and `gmtime()` and 2) the time is pre-UNIX Epoch.  That's an indication of a misfeature of MSVC; on a UN\*X, it'd show up as 1957-01-12 16:42:86 or something such as that, which is a valid-but-surprising date (but I know of at least one organization that used file system date fields to record events that long predated computers, so it's not inconceivable that they'd show up in SMB packets, given that they *did* show up in SMB packets).

Reply:
Precise time processing is a difficult subject, and it would be fine to let the code give up and fail safe on a strange value. But if you see it differently, make your try.

Reply:
OK, I've gotten rid of `nd_format_time_with_nanoseconds()` and also changed a bunch of error strings to indicate the underlying problem making it impossible to print the date/time in some reasonable form.

Reply:
That makes more sense to me, please note that the removed function still appears in `netdissect.h` and in the commit message. Speaking of which, the message also contains a "the the".

Reply:
Fixed.

Reply:
Thank you.